receiver/k8sclusterreceiver/internal/collection/nodes.go

```diff
@@ -61,27 +62,46 @@ func getMetricsForNode(node *corev1.Node, nodeConditionTypesToReport, allocatabl
 			},
 		})
 	}
+
 	// Adding 'node allocatable type' metrics
 	for _, nodeAllocatableTypeValue := range allocatableTypesToReport {
 		nodeAllocatableMetric := getNodeAllocatableMetric(nodeAllocatableTypeValue)
 		v1NodeAllocatableTypeValue := corev1.ResourceName(nodeAllocatableTypeValue)
 		metricValue, err := nodeAllocatableValue(node, v1NodeAllocatableTypeValue)
-
-		// metrics will be skipped if metric not present in node or value is not convertable to int64
-		if err != nil {
-			logger.Debug(err.Error())
+		val := utils.GetInt64TimeSeries(metricValue)
+		valType := metricspb.MetricDescriptor_GAUGE_INT64
+
+		if featuregate.IsEnabled(reportCPUMetricsAsDoubleFeatureGateID) {
+			metricQuantity, ok := node.Status.Allocatable[v1NodeAllocatableTypeValue]
+			if !ok {
+				logger.Debug(fmt.Errorf("allocatable type %v not found in node %v", nodeAllocatableTypeValue,
+					node.GetName()).Error())
+				continue
+			}
```

Comment:
`nodeAllocatableValue` is used only once in this function. Let's rather stop using the function than copying some parts of it

Reply:
Got it, done. 
When working with feature gates/toggles at previous gigs they often would prefer altering the original code as little as possible. I'll follow your advice now.